photon-client/src/main/scala/com/linkedin/photon/ml/data/DataValidators.scala

```diff
@@ -187,6 +189,19 @@ object DataValidators extends Logging {
   }
 
   /**
+   * Verify that a row has a positive weight.
+   *
+   * @param row The input row from a data frame
+   * @param inputColumnName The column name we want to validate
+   * @return Whether the weight of the input data point is positive
+   */
+  def rowHasPositiveWeight(row: Row, inputColumnName: String): Boolean = {
+    val weight = row.getAs[Double](inputColumnName)
```

Comment:
This will get a `ClassCastException` if the value isn't a `Double`. It would be better to do a `match` on this type, and return `false` if it doesn't match, e.g.:

```scala
row.getAs[Any](inputColumnName) match {
  case weight: Double => weight > MathConst.EPSILON
  case _ => false
}
```

Reply:
I'll make these changes in another commit or PR, since they're related to #300